internal/witness/cmd/witness/impl/witness.go

```diff
@@ -59,15 +60,28 @@ type ServerOpts struct {
 func buildLogMap(config LogConfig) (map[string]ct.SignatureVerifier, error) {
 	logMap := make(map[string]ct.SignatureVerifier)
 	for _, log := range config.Logs {
-		pk, err := ct.PublicKeyFromB64(log.PubKey)
+		// Use the PubKey string to first create the verifier.
+		der, err := base64.StdEncoding.DecodeString(log.PubKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode public key: %v", err)
+		}
+		pk, err := x509.ParsePKIXPublicKey(der)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create public key: %v", err)
 		}
 		logV, err := ct.NewSignatureVerifier(pk)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create signature verifier: %v", err)
 		}
-		logMap[log.LogID] = *logV
+		// And then to create the (alphanumeric) logID.
```

Comment:
It it worth extracting the common impl of this to be reused by feeder and witness?

Reply:
Sure, done!